bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/internal/values/XmlSequence.java

```diff
@@ -321,6 +321,9 @@ public void addChildren(BXml xmlItem) {
      */
     @Override
     public XmlValue strip() {
+        if (this.isFrozen()) {
```

Comment:
Strip create a new sequence out from this one, so we don't need to do this IMO

Reply:
does that mean that ```xml x5 = xml `<foo><bar/><?foo?></foo><?foo?>text1 text2<!--Com1--> <foo><bar/></foo>`;
  xml x2 = (x5.cloneReadOnly()).strip();``` should also be possible

Reply:
Compilation error is wrong as `strip` function is does not change the original value, it just create a new value skipping unwanted things.